tests/e2e/parallel/main_test.go

```diff
@@ -2795,3 +2795,40 @@ func TestScheduledSuiteTimeoutFail(t *testing.T) {
 		t.Fatal("The scan should have the timeout annotation")
 	}
 }
+
+func TestRuleHasProfileAnnotation(t *testing.T) {
+	t.Parallel()
+	f := framework.Global
+	const requiredRule = "ocp4-file-groupowner-worker-kubeconfig"
+	const expectedRuleProfileAnnotation = "ocp4-pci-dss-node,ocp4-moderate-node,ocp4-stig-node,ocp4-nerc-cip-node,ocp4-cis-node,ocp4-high-node"
+	err, found := framework.Global.DoesRuleExist(f.OperatorNamespace, requiredRule)
+	if err != nil {
+		t.Fatal(err)
+	} else if !found {
+		t.Fatalf("Expected rule %s not found", requiredRule)
+	}
+
+	// Check if requiredRule has the correct profile annotation
+	rule := &compv1alpha1.Rule{}
+	err = f.Client.Get(context.TODO(), types.NamespacedName{
+		Name:      requiredRule,
+		Namespace: f.OperatorNamespace,
+	}, rule)
+	if err != nil {
+		t.Fatal(err)
+	}
+	key, has := rule.Annotations[compv1alpha1.RuleProfileAnnotationKey]
+	if !has {
+		t.Fatalf("Expected rule %s to have the profile annotation", requiredRule)
+	}
+	// check the length of the annotation equals the expected length
+	if len(strings.Split(key, ",")) != len(strings.Split(expectedRuleProfileAnnotation, ",")) {
+		t.Fatalf("Expected rule %s to have the profile annotation %s but got %s", requiredRule, expectedRuleProfileAnnotation, key)
+	}
+	// loop over the expected profiles and check if the key contains them
+	for _, profile := range strings.Split(expectedRuleProfileAnnotation, ",") {
```

Comment:
We could implement the inverse of this assertion and use it in place of the previous check on line 2829.

Currently, we're checking the length is what we expect (e.g., does the result have six profiles). Then we're making sure each expected profile is in the result, asserting the sets are equal (the result shouldn't contain anything extra, it should have everything we expect).

We could generalize the containment check here and then flip the expected and actual order to perform both with a single assertion.

```go
func assertProfileInRuleAnnotation(r *compv1alpha1.Rule, expectedProfileId string) bool {
        profileIds := rule.Annotations[compv1alpha1.RuleProfileAnnotationKey]
        if strings.Contains(profileIds, expectedProfileId) {
               return true
        }
        return false
}

profiles := cmpv1alpha1.ProfileList{}
err := pcfg.Client.List(context.TODO(), &profileList,
	client.InNamespace(pcfg.ProfileBundleKey.Namespace),
	client.MatchingLabels{cmpv1alpha1.ProfileBundleOwnerLabel: pcfg.ProfileBundleKey.Name})

expectedProfiles := "..."
unexpectedProfiles := unique(profiles, expectedProfiles) // returns profiles that are not in expectedProfiles

for profile, _ in range expectedProfiles {
        if !assertProfileInRuleAnnotation(rule, profile) {
                t.Fatalf("expected to find profile %s in rule %s", profile, rule.Name)
        }
}

for profile, _ in range unexpectedProfiles {
        if !assertProfileInRuleAnnotation(rule, profile) {
                t.Fatalf("found unexpected profile %s in rule %s", profile, rule.Name)
        }
}
```

We could take this a step further and use it to remove the hardcoded expectation, which may break when a rule is removed from a particular profile in CaC/content.